cmd/lncli/commands.go

```diff
@@ -2298,14 +2305,32 @@ func sendPayment(ctx *cli.Context) error {
 
 	var rHash []byte
 
-	switch {
-	case ctx.IsSet("payment_hash"):
-		rHash, err = hex.DecodeString(ctx.String("payment_hash"))
-	case args.Present():
-		rHash, err = hex.DecodeString(args.First())
-		args = args.Tail()
-	default:
-		return fmt.Errorf("payment hash argument missing")
+	if ctx.Bool("key_send") {
+		if ctx.IsSet("payment_hash") {
+			return errors.New("cannot set payment hash when using " +
+				"key send")
+		}
+		var preimage lntypes.Preimage
+		if _, err := rand.Read(preimage[:]); err != nil {
+			return err
+		}
+
+		req.DestCustomRecords = map[uint64][]byte{
```

Comment:
should we have an integration test to assert this behavior?

Reply:
Do you mean an itest for `lncli`? Or an itest that tests the sender-receiver combination?

My view on itests is to only use them to cover things that cannot be covered in a unit test. It seems to me that every aspect of key send can be tested in isolation.

Reply:
Is there a particular interaction that you'd like to see covered?

Reply:
i think it'd be useful to cover the exact scenario that `lncli` feeds into the rpc, e.g. the correct custom records, dest, amt, payhash, etc

Reply:
Added